config/m4/compiler.m4

```diff
@@ -401,48 +413,76 @@ CXXFLAGS="$SAVE_CXXFLAGS"
 AC_LANG_POP
 AM_CONDITIONAL([HAVE_GNUXX11], [test "x$gnuxx11_happy" != xno])
 
-
 #
 # PGI specific switches
 #
 ADD_COMPILER_FLAG_IF_SUPPORTED([--display_error_number],
                                [--display_error_number],
-                               [AC_LANG_SOURCE([[int main(){return 0;}]])],
+                               [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
                                [],
                                [])
 
 # Suppress incorrect printf format for PGI18 compiler. TODO: remove it after compiler fix
 ADD_COMPILER_FLAG_IF_SUPPORTED([--diag_suppress 181],
                                [--diag_suppress 181],
-                               [AC_LANG_SOURCE([[int main(){return 0;}]])],
+                               [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
                                [],
                                [])
 
 # Suppress deprecated API warning for PGI18 compiler
 ADD_COMPILER_FLAG_IF_SUPPORTED([--diag_suppress 1215],
                                [--diag_suppress 1215],
-                               [AC_LANG_SOURCE([[int main(){return 0;}]])],
+                               [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
                                [],
                                [])
 
 # Use of a const variable in a constant expression is nonstandard in C
 ADD_COMPILER_FLAG_IF_SUPPORTED([--diag_suppress 1901],
                                [--diag_suppress 1901],
-                               [AC_LANG_SOURCE([[int main(){return 0;}]])],
+                               [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
                                [],
                                [])
 
 # Check if "-pedantic" flag is supported
 CHECK_COMPILER_FLAG([-pedantic], [-pedantic],
-                    [AC_LANG_SOURCE([[int main(){return 0;}]])],
+                    [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
                     [CFLAGS_PEDANTIC="$CFLAGS_PEDANTIC -pedantic"],
                     [])
 
+ADD_COMPILER_FLAGS_IF_SUPPORTED([[-Wno-missing-field-initializers],
```

Comment:
shall we use ADD_COMPILER_FLAGS_IF_SUPPORTED macro for other flags such as "--diag_suppress", "--display_error_number" etc?

Reply:
yep, good idea
done